ksql-engine/src/test/java/io/confluent/ksql/util/QueryMetadataTest.java

```diff
@@ -16,81 +16,110 @@
-  private final Map<String, Object> overriddenProperties = Collections.emptyMap();
+  private static final String QUERY_APPLICATION_ID = "Query1";
 
+  @Mock
+  private OutputNode outputNode;
+  @Mock
+  private KafkaTopicClient kafkaTopicClient;
+  @Mock
+  private Topology topoplogy;
+  @Mock
   private KafkaStreams kafkaStreams;
+  @Mock
+  private QueryStateListener listener;
   private Metrics metrics;
   private MetricName metricName;
-  private final String metricGroupName = "ksql-queries";
+  private QueryMetadata query;
 
   @Before
   public void setup() {
     metrics = MetricsTestUtil.getMetrics();
-    metricName = metrics.metricName("query-status", metricGroupName,
+    metricName = metrics.metricName("query-status", "ksql-queries",
         "The current status of the given query.",
-        Collections.singletonMap("status", queryApplicationId));
-    outputNode = EasyMock.niceMock(OutputNode.class);
-    kafkaStreams = EasyMock.niceMock(KafkaStreams.class);
-  }
+        Collections.singletonMap("status", QUERY_APPLICATION_ID));
 
-  @Test
-  public void shouldAddandRemoveTheMetricOnClose() {
-    EasyMock.expect(kafkaStreams.state()).andReturn(State.RUNNING).once();
-    EasyMock.expect(kafkaStreams.state()).andReturn(State.NOT_RUNNING).once();
-    EasyMock.replay(kafkaStreams, outputNode);
-    final QueryStateListener queryStateListener = new QueryStateListener(metrics, kafkaStreams, queryApplicationId);
-    kafkaStreams.setStateListener(EasyMock.eq(queryStateListener));
-    final QueryMetadata queryMetadata = new QueryMetadata(
-        statementString,
+    query = new QueryMetadata(
+        "foo",
         kafkaStreams,
         outputNode,
-        executionPlan,
-        dataSourceType,
-        queryApplicationId,
+        "bar",
+        DataSourceType.KSTREAM,
+        QUERY_APPLICATION_ID,
         kafkaTopicClient,
         topoplogy,
-        overriddenProperties
+        Collections.emptyMap()
     );
-    queryMetadata.registerQueryStateListener(queryStateListener);
-    queryMetadata.start();
-    assertThat(metrics.metric(metricName).metricName().name(), equalTo("query-status"));
-    assertThat(metrics.metric(metricName).metricValue().toString(), equalTo("RUNNING"));
-    queryStateListener.onChange(State.REBALANCING, State.RUNNING);
-    assertThat(metrics.metric(metricName).metricValue().toString(), equalTo("REBALANCING"));
-    queryStateListener.onChange(State.RUNNING, State.REBALANCING);
-    assertThat(metrics.metric(metricName).metricValue().toString(), equalTo("RUNNING"));
-    queryMetadata.close();
-    EasyMock.verify(kafkaStreams);
-    assertThat(metrics.metric(metricName), nullValue());
+  }
+
+  @Test
+  public void shouldSetInitialStateWhenListenerAdd() {
+    // Given:
+    when(kafkaStreams.state()).thenReturn(State.CREATED);
+
+    // When:
+    query.registerQueryStateListener(listener);
+
+    // Then:
+    verify(listener).onChange(State.CREATED, State.CREATED);
```

Comment:
Why are both new and old stats the same?

Reply:
Because its just pushing the initial state.  So there is no previous state.